PE-19086 Change extend gpg key to ignore gpg key warnings

cthorn42 opened the pull request:
The extended GPG-key for PE 2015.2 -> 2016.1.2 has expired.
Rather then continue to update it every few months, lets just
enable the older versions of PE that are installed on debian
systems to ignore gpg-key warnings.
We will still have testing coverage of the gpg-key on debian
systems via fresh installs of PE and testing upgrades from
PE 2016.2.0 and newer.

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/145/

cthorn42 commented:
Looks like I was wrong about the lower end, PE 3.8.5, and PE 3.8.6 both now come with an expired key. Updated the method to make sure if it is those versions to also add in the ignore gpg-key warning, also updated specs to reflect the new behavior.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/146/

cthorn42 commented:
Bah humbug. Looks like PE 2016.2.1 *also* has an expired gpg-key. I need to do more research to really nail that down.

cthorn42 commented:
Okay updated everything to reflect now every PE version less then 2016.4.0 needs this ignore method if it is deb/ubuntu. 
2016.4.0 and up has a gpg key that expires in 2021, we should be good until then.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/147/

tvpartytonight commented:
👍

kevpl commented:
👍
